packages/x-components/src/x-modules/scroll/components/main-scroll.vue

```diff
@@ -0,0 +1,400 @@
+     * `id` property, this component should be scrolled into view.
+     *
+     * @internal
+     */
+    @State('scroll', 'pendingScrollTo')
+    public pendingScrollTo!: string;
+
+    /**
+     * Intersection observer to determine visibility of the elements.
+     *
+     * @returns An intersection observer to detect elements visibility.
+     * @internal
+     */
+    protected intersectionObserver: IntersectionObserver | null = null;
+
+    /**
+     * Creates an `IntersectionObserver` to detect the first visible elements. Children of this
+     * component should register themselves if they want to be observed.
+     *
+     * @returns The intersection observer.
+     * @public
+     */
+    @XProvide(FirstVisibleItemObserverKey)
+    public get firstVisibleElementObserver(): ScrollVisibilityObserver | null {
+      const observer = this.intersectionObserver;
+      return observer
+        ? {
+            observe: observer.observe.bind(observer),
+            unobserve: element => {
+              this.removeVisibleElement(element);
+              observer.observe(element);
+            }
+          }
+        : null;
+    }
+
+    /**
+     * The first visible element contained in this component.
+     *
+     * @returns The first visible element in this component.
+     * @internal
+     */
+    protected get firstVisibleElement(): HTMLElement | null {
+      return this.intersectingElements.length === 0
+        ? null
+        : this.intersectingElements.reduce((firstVisibleElement, anotherElement) => {
+            // FIXME: This algorithm only takes into account LTR layouts
+            const firstVisibleElementBounds = firstVisibleElement.getBoundingClientRect();
+            const anotherElementBounds = anotherElement.getBoundingClientRect();
+            return anotherElementBounds.left <= firstVisibleElementBounds.left &&
+              anotherElementBounds.top <= firstVisibleElementBounds.top
+              ? anotherElement
+              : firstVisibleElement;
+          });
+    }
+
+    /**
+     * Initialise the observer after mounting the component.
+     */
+    mounted(): void {
+      this.intersectionObserver = new IntersectionObserver(this.updateVisibleElements, {
+        root: this.useWindow ? undefined : this.$el,
+        threshold: this.threshold,
+        rootMargin: this.margin
+      });
+    }
+    /**
+     * Disconnects the intersection observer.
+     *
+     * @internal
+     */
+    beforeDestroy(): void {
+      this.intersectionObserver?.disconnect();
+      this.$x.emit('UserScrolledToElement', '');
+    }
+
+    /**
+     * Emits the first visible element `[data-scroll]`.
+     *
+     * @param element - The first visible element or `null` if there aren't any.
+     * @internal
+     */
+    @Watch('firstVisibleElement')
+    protected emitFirstVisibleElement(element: HTMLElement | null): void {
+      this.$x.emit('UserScrolledToElement', element?.dataset.scroll ?? '');
```

Comment:
Is it possible to check if the scroll element is at the start (scrollTop===0) and avoid to emit the event. Also clear the current scroll from URL when scroll top is reached?

Because when you search, the first id is set on URL even when the scrollTop is 0. It would be great to avoid that.

Reply:
right now if you search, and reload, we have the issue of the animation and the grid is scrolled a little. We can try to avoid at least if the scroll is on top